org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/configuration/internal/BuildConfigurationPersistence.java

```diff
@@ -108,33 +110,47 @@ public void deletePathToRoot(File projectDir) {
         deleteRootDirPreference(preferences);
     }
 
-    private static BuildConfigurationProperties readPreferences(PreferenceStore preferences, File rootDir) {
-        String distribution = preferences.readString(PREF_KEY_CONNECTION_GRADLE_DISTRIBUTION, null);
-        if (distribution == null) {
-            throw new GradlePluginsRuntimeException("Invalid build configuration for project located at " + rootDir.getAbsolutePath());
-        }
+    private static DefaultBuildConfigurationProperties readPreferences(PreferenceStore preferences, File rootDir) {
         boolean overrideWorkspaceSettings = preferences.readBoolean(PREF_KEY_OVERRIDE_WORKSPACE_SETTINGS, false);
+
+        String distributionString = preferences.readString(PREF_KEY_CONNECTION_GRADLE_DISTRIBUTION, null);
+        GradleDistribution distribution = distributionString == null
+                ? GradleDistribution.fromBuild()
+                : GradleDistributionSerializer.INSTANCE.deserializeFromString(distributionString);
+
+        String gradleUserHomeString = preferences.readString(PREF_KEY_GRADLE_USER_HOME, "");
```

Comment:
I think only in the project config is enough for now, because it feels more surprising there. I'd only show it when the user actually configures a path.